xwiki-platform-core/xwiki-platform-security/xwiki-platform-security-requiredrights/xwiki-platform-security-requiredrights-rest/src/main/java/org/xwiki/platform/security/requiredrights/rest/internal/AvailableRightsManager.java

```diff
@@ -0,0 +1,152 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.platform.security.requiredrights.rest.internal;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+import org.xwiki.component.annotation.Component;
+import org.xwiki.localization.ContextualLocalizationManager;
+import org.xwiki.model.EntityType;
+import org.xwiki.model.reference.DocumentReference;
+import org.xwiki.platform.security.requiredrights.RequiredRight;
+import org.xwiki.platform.security.requiredrights.RequiredRightAnalysisResult;
+import org.xwiki.security.authorization.AuthorizationManager;
+import org.xwiki.security.authorization.Right;
+import org.xwiki.security.authorization.requiredrights.DocumentRequiredRight;
+import org.xwiki.security.requiredrights.rest.model.jaxb.AvailableRight;
+import org.xwiki.security.requiredrights.rest.model.jaxb.ObjectFactory;
+import org.xwiki.user.CurrentUserReference;
+import org.xwiki.user.UserReferenceSerializer;
+
+/**
+ * Compute the available rights for the current user on a given document.
+ *
+ * @version $Id$
+ * @since 17.4.0RC1
+ */
+@Component(roles = AvailableRightsManager.class)
+@Singleton
+public class AvailableRightsManager
+{
+    private static final List<DocumentRequiredRight> CONSIDERED_RIGHTS = List.of(
+        new DocumentRequiredRight(null, EntityType.DOCUMENT),
```

Comment:
Might worth a comment: while performing a quick review I'm not sure to understand what this DocumentRequiredRight taking a null value is.

Reply:
This is the "None" entry, I'll add a comment.

Reply:
Yeah I finally figured it out, but not immediately.

Reply:
I've added a comment.